src/platform_impl/web/web_sys/canvas.rs

```diff
@@ -87,9 +122,25 @@ impl Canvas {
         self.raw.height() as f64
     }
 
-    pub fn set_size(&self, size: LogicalSize) {
+    pub fn set_size(&self, size: LogicalSize, intended: bool) {
         self.raw.set_width(size.width as u32);
         self.raw.set_height(size.height as u32);
+
+        if intended {
+            *self.intended_size.borrow_mut() = size;
+        }
+    }
+
+    pub fn intended_size(&self) -> LogicalSize {
+        *self.intended_size.borrow()
+    }
+
+    pub fn window_size(&self) -> LogicalSize {
+        let window = web_sys::window().expect("Failed to obtain window");
+        let width = window.inner_width().expect("Failed to get width").as_f64().expect("Failed to get width as f64");
+        let height = window.inner_height().expect("Failed to get height").as_f64().expect("Failed to get height as f64");
+
+        LogicalSize { width, height }
     }
```

Comment:
This should probably be a backend static method: `web_sys::window_size()`.

Reply:
👍 Good point